Library/Homebrew/cmd/update-report.rb

```diff
@@ -617,15 +617,13 @@ def dump_formula_or_cask_report(key, title)
       # Format list items of formulae
       case key
       when :R
-        if report_all
```

Comment:
Presumable this is removed because this is only ever reached when `Homebrew::EnvConfig.update_report_all_formulae?` is set? If so, I think it's worth doing a bit more refactoring to make that more apparent maybe?

Reply:
> We could keep the redundant check for readability, maybe? In that case, I think just adding a guard clause to `dump_formula_or_cask_report` would make more sense than having a check inside the loop.

Yeh, either of these would make sense to me 👍🏻

Reply:
Another refactoring option is to split this `case` into multiple methods given they all operate independently on the data.

Reply:
> Another refactoring option is to split this `case` into multiple methods given they all operate independently on the data.

I think I like this option the best.